components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -1011,6 +1011,72 @@ public static String getQueryStringWithFrameworkContextId(String originalQuerySt
         return idpAuthenticatorMap;
     }
 
+    // IDS_2192 - Add a method to retrieve RequestPath Authenticators who have previously authenticated the user
+    public static Map<String, AuthenticatorConfig> getAuthenticatedRequestPathIdPs(SequenceConfig sequenceConfig, Map<String, AuthenticatedIdPData> authenticatedIdPs) {
+        if (log.isDebugEnabled()) {
+            log.debug("Finding already authenticated request path IdPs");
+        }
+
+        Map<String, AuthenticatorConfig> idpAuthenticatorMap = new HashMap<String, AuthenticatorConfig>();
+        List<AuthenticatorConfig> requestPathAuthenticators = sequenceConfig.getReqPathAuthenticators();
```

Comment:
This will work only if the same request path authenticator has been configured for each service provider that will be participating in the user session.
So that, when the first application uses request path authentication, the second application which will use the basic authentication scheme, but also has request path authentication option being configured will work.
If the request path authenticator has not been configured for the second application, it should still prompt for credentials in this fix.

We need to clearly validate the user story for this usecase.  That is whether the user should get SSO for an application that will only use basic authentication scheme, when the user is already logged into an application over request path basic authentication scheme.

As I see, with this fix, additional configurations needs to be done to achieve above.